src/core/tensor/tensor.cc

```diff
@@ -152,29 +152,24 @@ Tensor Resize(const Tensor &in, const Shape &shape) {
     }                                                                          \
   } while (0)
 
-Tensor &Tensor::AsType(const DataType type) {
+// return new tensor
+Tensor Tensor::AsType(const DataType type) {
```

Comment:
do we need to add a flag for in-place? default can be False.

Reply:
Yes. We could add a flag to indicate this explictly. But second thought is, by adding in-place flag, `in-place=True` is also expected to be available. Then, it go back to the previous discussion, if we want to do `in-place` for multi-referenced tensor, what is the expected behavior?

Reply:
Do we have `in-place` for any other tensor functions?
the behavior is 
1. in-place = true, then reuse the memory (only works if the bitwidth is the same for the two data types).
2. in-place= false, return a tensor with a new memory block.

case 1 is a bit confusing for the case that the bitwidths are different for the two data types although it may save some cost.

so let's just ignore the in-place flag.

is your concern in the same?

Reply:
Yes. this is my concern.

Currently the `in place` flag are available in python tensor api. No one in cpp tensor api yet.